packages/lodestar/src/network/util.ts

```diff
@@ -81,3 +82,15 @@ export function eth2ResponseTimer<T>(): (source: AsyncIterable<T>) => AsyncGener
     })();
   };
 }
+
+export async function dialProtocol(libp2p: LibP2p, peerId: PeerId, protocol: string, timeout: number): Promise<Stream> {
+  let connectionTimer;
+  const {stream} = await Promise.race([
+    libp2p.dialProtocol(peerId, protocol),
```

Comment:
you aren't using abort signal (I think abort signal will reject dial protocol anyways)

Reply:
so create new separate AbortController here and put its signal to `dialProtocol` and do `controller.abort()` in `setTimeout`?

Reply:
yes